### PR TITLE
Fix ChanelShuffle named tensor warnings

### LINKED ISSUE #54846: Irrelevant named tensor warnings
## 🐛 Bug
This code:
```
import torch
import torch.nn as nn

input_tensor = torch.ones((1, 32, 128, 128), dtype=torch.float)
cs = nn.ChannelShuffle(groups=2)
cs(input_tensor)
```

produces irrelevant warning:

> /home/user/PyTorch/dist-packages/release/torch/nn/modules/channelshuffle.py:46: UserWarning: Named tensors and all their associated APIs are an experimental feature and subject to change. Please do not use them for anything important until they are released as stable. (Triggered internally at  ../c10/core/TensorImpl.h:973.)
>   return F.channel_shuffle(input, self.groups)

as no named tensors are used.

## Expected behavior

No warnings

## Environment

 - PyTorch Version (e.g., 1.0): current master
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): source

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @albanD @mruberry

### PR DESCRIPTION
**Pull request by h6197627**
Fixes #54846

### CODE REVIEW

File: aten/src/ATen/native/ChanelShuffle.cpp [hunk 1]
```diff
@@ -54,7 +54,7 @@ Tensor channel_shuffle(const Tensor& self, int64_t groups) {
       .reshape(self.sizes());
   return namedinference::propagate_names_if_nonempty(
       output_tensor,
-      self.names());
+      self.has_names() ? self.names() : std::vector<Dimname>());
```
Comment (zou3519) [inline aten/src/ATen/native/ChanelShuffle.cpp:57]: Can you add a test somewhere to confirm this solves the problem? Maybe in `test/test_nn.py` or `test/test_named_tensor`

Comment (h6197627) [reply aten/src/ATen/native/ChanelShuffle.cpp:57]: Yes, I will try. Maybe it will take some time as I am not familiar with PyTorch testing

Comment (zou3519) [reply aten/src/ATen/native/ChanelShuffle.cpp:57]: Here's an example of how to test for the existence of no warnings:
https://github.com/pytorch/pytorch/blob/75eb026e07062e8539c3b6e96d6100ac9378a292/test/test_torch.py#L2898-L2902

Comment (ezyang) [inline aten/src/ATen/native/ChanelShuffle.cpp:57]: `at::ArrayRef<Dimname>{}` will work here and is more efficient

Comment (jbschlosser) [reply aten/src/ATen/native/ChanelShuffle.cpp:57]: @h6197627 You can add the no-warning check to the existing `channel_shuffle` test here:
https://github.com/pytorch/pytorch/blob/ba320cec6bc0e20855f8f6ce74f4248dcfafff5e/test/test_nn.py#L9159

Comment (h6197627) [reply aten/src/ATen/native/ChanelShuffle.cpp]: @zou3519, @jbschlosser, I added generic checking for all warnings in ChannelShuffle. However, if it is not appropriate I can change it to something more specific like:
`self.assertNotIn('Named tensors and all their associated APIs are an experimental feature and subject to change.', ' '.join(map(str, w)))`

**Review by jbschlosser — APPROVED** [review]
LGTM! Thanks for the fix.

**Pull request closed without merging**